### ISSUE TITLE
Handling of non-ascii usernames

### ISSUE THREAD

**Issue by brutasse** [django-webtest/django-webtest#8]
tl;dr: replace `'foo'` with `'foò'` in the test suite and watch things break.

Non-ascii usernames seems a bit broken at the moment, at least on python2. Since `to_string` (from compat.py) calls `str()` on the value if it's unicode, this fails with an encoding error when the given value is not plain ascii.

`self.app.get(url, user=u'foò')` breaks completely on python 2. The workaround is to pass a native string instead of a unicode string with `self.app.get(url, user=force_bytes(username))` everywhere. This somewhat works but the test output ends up with lots `UnicodeWarnings` and the client code isn't very friendly. This also prevents using a user instance as `user=` kwarg since `get_username()` returns a unicode string instead of a native string.

The WSGI pep says native strings must be used but they're not the same thing in py2 and py3.

I've tried to come up with an implementation that doesn't break on unicode based on what [webtest does](https://github.com/Pylons/webtest/blob/master/webtest/lint.py#L140-L144) data but failed so far.

Ideas?
